azkaban-common/src/main/java/azkaban/executor/ExecutionOptions.java

```diff
@@ -166,6 +172,52 @@ public static ExecutionOptions createFromObject(final Object obj) {
     return options;
   }
 
+  /**
+   * Validate the ExecutionOption's Flow-Parameters against the application level Properties
+   * @throws ServletException if any of the parameter is invalid
+   */
+  public void validateFlowParameters(Props azProps)
+      throws ServletException {
+    List<String> errMsg = new ArrayList<>();
+
+    Map<String, String> flowParameters = this.getFlowParameters();
+    if (flowParameters == null || flowParameters.isEmpty()) {
+      return;
+    }
+    if (flowParameters.containsKey(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS)) {
+
+      // user defined list
+      final String[] statuses = flowParameters
+          .getOrDefault(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS, "")
+          .split("\\s*,\\s*");
+
+      for (String s : statuses) {
+        if (!RESTARTABLE_TERMINATED_STATUSES.contains(Status.valueOf(s))) {
+          errMsg.add(String.format("`%s` is not a valid restartable status, "
+              + "permitted status are %s", s, RESTARTABLE_TERMINATED_STATUSES));
+        }
+      }
+    }
+    if (flowParameters.containsKey(FlowParameters.FLOW_PARAM_RESTART_COUNT)){
+      // check restart count limit
+      final int flowRestartCountLimit = azProps.getInt(
+          AZKABAN_EXECUTION_RESTART_LIMIT, DEFAULT_FLOW_RESTART_LIMIT);
+      final int flowRestartCount = Integer.parseInt(
+          flowParameters.getOrDefault(FlowParameters.FLOW_PARAM_RESTART_COUNT, "0"));
```

Comment:
I think parseInt can throw NumberFormatException. When/how are capturing that exception? What will the user see?

Reply:
It's in the screenshot, will see at flow submit time

Reply:
I think @ypadron-in meant if they pass a value which is not a number, like ,'x' or '1x' or something similar. 
parstInt will throw an exception then will it not ? The above will cover cases if a wrong number is passed.

Reply:
oh I see. let me add this

Reply:
thanks for adding it in a try block.